src/KKManager/Windows/Content/CardWindow.cs

```diff
@@ -873,10 +867,225 @@ private void openInExplorerToolStripMenuItem_Click(object sender, EventArgs e)
+        }
+
+        private void selectMarkedToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                if (!listView.CheckBoxes)
+                {
+                    MessageBox.Show("Enable 'Use checkboxes' in View first.", "Select marked", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                    return;
+                }
+                var checkedItems = _typedListView.CheckedObjects;
+                if (checkedItems == null || checkedItems.Count == 0)
+                {
+                    MessageBox.Show("No marked items.", "Select marked", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                    return;
+                }
+                listView.SelectObjects(checkedItems.Cast<object>().ToList());
+                listView.FocusedObject = checkedItems[0];
+            }
+            catch { }
+        }
+
+        private void moveSelectedToFolderToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            var targets = GetActionTargets();
+            if (targets == null || targets.Count == 0)
+            {
+                MessageBox.Show("Nothing selected.", "Move cards", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                return;
+            }
+            using (var dlg = new FolderBrowserDialog())
+            {
+                dlg.Description = "Choose destination folder";
+                if (dlg.ShowDialog(this) != DialogResult.OK) return;
+                var destRoot = new DirectoryInfo(dlg.SelectedPath);
+                if (!destRoot.Exists)
+                {
+                    MessageBox.Show("Destination folder does not exist.", "Move cards", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+                    return;
+                }
+                int moved = 0, failed = 0;
+                foreach (var card in targets)
+                {
+                    try
+                    {
+                        var destPath = Path.Combine(destRoot.FullName, card.Location.Name);
+                        var candidate = destPath;
+                        int n = 1;
+                        while (File.Exists(candidate))
+                        {
+                            var name = Path.GetFileNameWithoutExtension(destPath);
+                            var ext = Path.GetExtension(destPath);
+                            candidate = Path.Combine(destRoot.FullName, $"{name} ({n++}){ext}");
+                        }
+                        File.Move(card.Location.FullName, candidate);
+                        moved++;
+                    }
+                    catch { failed++; }
+                }
+                MessageBox.Show($"Moved {moved} file(s){(failed > 0 ? ", some failed" : "")}.", "Move cards", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                RefreshList();
+            }
+        }
+
+        private void RebuildDuplicateFilter()
```

Comment:
If the filtering logic was moved into the ModelFilter then rebuilding would become unnecessary, checking the toggles would just need to do a quick list refresh. Performance-wise it should be faster for when cards are still being actively loaded.